### ISSUE TITLE
AWS::Serverless::SimpleTable - Incorrect Property Name Results in Unhandled Exception

### ISSUE THREAD

**Issue by mlaff** [awslabs/serverless-application-model#884]
**Description:**

When using resource `AWS::Serverless::SimpleTable`, if you use the name `Id` in `PrimaryKey` property instead of the correct `Name`, and try to deploy, the exception is unhandled, but it should be an InvalidResourceException.

**Steps to reproduce the issue:**
1. Create template with content:
```
AWSTemplateFormatVersion: '2010-09-09'
Resources:
  SomeTable:
    Properties:
      PrimaryKey:
        Id: id
        Type: String
    ProvisionedThroughput:
      ReadCapacityUnits: 1
      WriteCapacityUnits: 1
    TableName: SomeTable
    Type: AWS::Serverless::SimpleTable
Transform: AWS::Serverless-2016-10-31
```
2. Try to deploy

**Observed result:**
Unhandled exception

**Expected result:**
InvalidResourceException

**Comment by keetonian** [comment]
Steps to fix this:

1. Add validation to https://github.com/awslabs/serverless-application-model/blob/master/docs/globals.rst#unsupported-properties to make sure `Name` exists as a property of PrimaryKey
1. Add tests to ensure this error is caught in https://github.com/awslabs/serverless-application-model/blob/develop/tests/translator/test_translator.py#L444

**Comment by jlhood** [comment]
Pending v1.14.0 release

**Comment by ShreyaGangishetty** [comment]
Closing this issue as this is released in v1.14.0
